zcash_client_backend/src/keys.rs

```diff
@@ -537,7 +548,13 @@ impl UnifiedFullViewingKey {
         #[cfg(not(feature = "transparent-inputs"))]
         let transparent = None;
 
-        UnifiedAddress::from_receivers(None, sapling, transparent)
+        let orchard = self
+            .orchard
+            .as_ref()
+            .filter(|_| receiver_types.contains(&Typecode::Orchard))
+            .map(|ofvk| ofvk.address_at(j, Scope::External));
+
+        UnifiedAddress::from_receivers(orchard, sapling, transparent)
```

Comment:
I would suggest putting the Orchard part first so that the code is in receiver priority order (like the arguments to `from_receivers`).